### PR TITLE
DROID-1925 Types | Fix | Extend search api to be able to include several spaces when needed

### PR DESCRIPTION
**Pull request by uburoiubu**

### CODE REVIEW

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt [hunk 1]
```diff
@@ -111,7 +111,10 @@ class ObjectTypeChangeViewModelTest {
         stubSpaceManager(spaceId)
         val vm = givenViewModel()
 
-        val expectedMyTypesFilters = ObjectSearchConstants.filterTypes(spaceId, SupportedLayouts.editorLayouts)
+        val expectedMyTypesFilters = ObjectSearchConstants.filterTypes(
+            spaces = listOf(spaceId),
+            SupportedLayouts.editorLayouts
```
Comment (konstantiniiv) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt:116]: recommendedLayouts =

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt [hunk 2]
```diff
@@ -150,14 +153,17 @@ class ObjectTypeChangeViewModelTest {
...
```
Comment (konstantiniiv) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt:158]: recommendedLayouts =

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt [hunk 3]
```diff
@@ -241,14 +247,17 @@ class ObjectTypeChangeViewModelTest {
...
```
Comment (konstantiniiv) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt:252]: recommendedLayouts =

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt [hunk 4]
```diff
@@ -410,14 +419,17 @@ class ObjectTypeChangeViewModelTest {
         val expectedInstalledTypeId = MockDataFactory.randomUuid()
         val expectedInstalledTypeUniqueKey = ObjectTypeIds.PAGE
 
-        val expectedMyTypesFilters = ObjectSearchConstants.filterTypes(space, SupportedLayouts.editorLayouts)
+        val expectedMyTypesFilters = ObjectSearchConstants.filterTypes(
+            spaces = listOf(space),
+            SupportedLayouts.editorLayouts
```
Comment (konstantiniiv) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt:424]: recommendedLayouts =

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt [hunk 5]
```diff
@@ -556,7 +568,10 @@ class ObjectTypeChangeViewModelTest {
         stubSpaceManager(spaceId)
         val vm = givenViewModel()
 
-        val expectedMyTypesFilters = ObjectSearchConstants.filterTypes(spaceId, SupportedLayouts.editorLayouts)
+        val expectedMyTypesFilters = ObjectSearchConstants.filterTypes(
+            spaces = listOf(spaceId),
+            SupportedLayouts.editorLayouts
```
Comment (konstantiniiv) [inline presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt:573]: recommendedLayouts =

**Review by konstantiniiv — APPROVED** [review]

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt (hunk 3 above)
Comment (uburoiubu) [reply presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt]: fixed

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt (hunk 4 above)
Comment (uburoiubu) [reply presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt]: fixed

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt (hunk 5 above)
Comment (uburoiubu) [reply presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt]: fixed

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt (hunk 2 above)
Comment (uburoiubu) [reply presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt]: fixed

File: presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt (hunk 1 above)
Comment (uburoiubu) [reply presentation/src/test/java/com/anytypeio/anytype/presentation/types/ObjectTypeChangeViewModelTest.kt]: fixed

**Pull request merged**